### ISSUE TITLE
Creating pipe Too many open files on windows 10

### ISSUE THREAD

**Issue by GreyZhang** [redguardtoo/elpa-mirror#38]
![image](https://user-images.githubusercontent.com/11857400/114379374-51e3b800-9bbb-11eb-9486-7b4c35c6ab10.png)

Creating pip Too many open files errors are eported on windows 10, I think it were because read elpa mirror is too fast and too many elisp files were being compiled at the same time.

All packages would be installed by restaring emacs for a second time.

All the packages:
ac-ispell-20151101.226
ace-jump-helm-line-20160918.1836
ace-link-20191118.935
ace-window-0.9.0
adaptive-wrap-0.7
ag-20190726.9
aggressive-indent-1.8.3
alert-20190816.2205
anzu-20190303.1701
archives
async-20200809.501
auto-compile-20191020.1040
auto-complete-20170125.245
auto-highlight-symbol-20130313.943
auto-yasnippet-20191015.942
avy-0.5.0
bind-key-20191110.416
bind-map-20161207.1511
clean-aindent-mode-20171017.2043
column-enforce-mode-20171030.1900
company-20191114.1356
company-statistics-20170210.1933
company-web-20180402.1155
dash-20200803.1520
deferred-20170901.1330
define-word-20191031.1736
diff-hl-20200701.2141
diminish-20170419.1736
dumb-jump-20191113.503
elisp-slime-nav-20160128.1909
elpa-mirror-20210325.1219
emmet-mode-20180613.341
epl-20180205.2049
eval-sexp-fu-20190109.809
evil-20191121.752
evil-anzu-20170124.718
evil-args-20180908.2157
evil-ediff-20170724.1923
evil-escape-20180910.1234
evil-exchange-20191105.255
evil-iedit-state-20180607.558
evil-indent-plus-20151109.1906
evil-lisp-state-20160404.248
evil-magit-20200604.110
evil-matchit-20191023.2322
evil-mc-20190916.348
evil-nerd-commenter-20191028.115
evil-numbers-20140606.1251
evil-search-highlight-persist-20170523.334
evil-surround-20191013.1656
evil-tutor-20150103.650
evil-unimpaired-0.1pre0.20210412.174325
evil-visual-mark-mode-20190116.1557
evil-visualstar-20160223.48
exec-path-from-shell-20190426.2227
expand-region-20190416.538
eyebrowse-20190928.1458
f-20191110.1357
fancy-battery-20150101.1204
fill-column-indicator-20191109.10
find-file-in-project-20201111.407
flx-20191115.659
flx-ido-20180117.1519
fringe-helper-20140620.2109
fuzzy-20150730.337
git-commit-20200828.1753
git-gutter+-20151204.1723
git-gutter-20201026.850
git-gutter-fringe+-20140729.1103
git-gutter-fringe-20200323.2249
git-link-20200827.2326
git-messenger-20200321.2337
git-timemachine-20200603.701
gitattributes-mode-20180318.1956
gitconfig-mode-20180318.1956
gitignore-mode-20180318.1956
gntp-20141025.250
gnuplot-20141231.2137
golden-ratio-20191028.1732
google-translate-20190620.1416
goto-chg-20190110.2114
haml-mode-20190219.2102
helm-20191122.1454
helm-ag-20170209.1545
helm-c-yasnippet-20170128.1542
helm-company-20190812.1429
helm-core-20191121.2004
helm-css-scss-20140627.25
helm-descbinds-20190501.935
helm-flx-20180103.516
helm-gitignore-20170211.8
helm-make-20190729.1221
helm-mode-manager-20151124.938
helm-projectile-20190731.1538
helm-swoop-20191028.1402
helm-themes-20160918.545
highlight-20190710.1527
highlight-indentation-20181204.839
highlight-numbers-20181013.1744
highlight-parentheses-20180704.1102
hl-todo-20191105.2239
htmlize-20191111.2130
hungry-delete-20170412.102
hydra-20191121.936
iedit-20190419.803
indent-guide-20191106.240
ivy-20201214.2049
link-hint-20190721.1844
linum-relative-20180124.1047
log4e-20170401.1304
lorem-ipsum-20190819.2042
lv-20191106.1238
macrostep-20161120.2106
magit-20201028.1912
magit-gitflow-20170929.824
magit-popup-20200719.1015
monokai-theme-20190801.1701
move-text-20170909.330
neotree-20181121.2026
open-junk-file-20161210.1114
org-bullets-20190802.927
org-category-capture-20180601.242
org-download-20191118.934
org-mime-20191106.1140
org-plus-contrib-20190909
org-pomodoro-20190530.1445
org-present-20180303.2330
org-projectile-20190130.1439
orgit-20200714.1943
packed-20180318.1729
paradox-20191011.1111
parent-mode-20150824.2300
pcre2el-20161120.2103
persp-mode-20190511.1402
pkg-info-20150517.1143
popup-20160709.1429
popwin-20150315.1300
powerline-20190323.213
powershell-20201005.1642
projectile-20191024.721
pug-mode-20180513.2126
rainbow-delimiters-20191018.1233
request-20191122.1416
restart-emacs-20180601.1031
s-20180406.808
sass-mode-20190502.53
scss-mode-20180123.1708
slim-mode-20170728.1348
smartparens-20191015.1754
smeargle-20200323.533
sound-wav-20181126.1726
spaceline-20181223.2024
spinner-1.7.3
tagedit-20161121.855
toc-org-20190902.1055
transient-20201020.1707
undo-tree-0.6.5
use-package-20191110.416
uuidgen-20140918.2301
vi-tilde-fringe-20141028.242
volatile-highlights-20160612.155
web-completion-data-20160318.848
web-mode-20191104.1859
which-key-20191111.1951
winum-20190911.1607
with-editor-20201026.2027
ws-butler-20170111.2334
yasnippet-20191030.1331

**Comment by redguardtoo** [comment]
abc8d7b run tar on windows (Chen Bin)

**Comment by GreyZhang** [comment]
After use the lastest version the problem still exists. 
I don't understand elisp, and I tried to read the changes in the file. I found some information about cygwin and I did use cygwin and add bin directory into Path. The installation directory is root of C disk.
I also checked for tar executable files and found there are two files on my computer. One is in C:\Windows\System32, version information: bsdtar 3.3.2 - libarchive 3.3.2 zlib/1.2.5.f-ipp. The other is tar (GNU tar) 1.29. I tried to remove the bsdtar version yet the problem still exists.

**Comment by redguardtoo** [comment]
Both bsd tar and gnu tar should be ok. 

I add Cygwin related code because I'm using windows 7.

Could you give me exact steps to reproduce the issue?

**Comment by GreyZhang** [comment]
Ok, I just found a windows7 x64 computer and tried to use your emacs configuration to have a test.  Emacs version: 26.1.

Steps:
1. I tried to use the stable version, yet found there is a problem when installing, nov package need esxml 0.3.6, yet only 0.3.5 is available. So I forced to change the archive-contents information to install it with esxml 0.3.5(not know how to solve it, yet it is not the issue we are talking about).
2. uncomment myelpa configurations in .emacs.d/lisp/init-elpa.el
3. Generate mirror of elpa into ~/myelpa, 242 files in all in this folder after generation.
4. Reove all folders and files under .emacs.d/elpa, just keep elpa-mirror-2.1.4 and cmake-mode-3.20.0pre5(using elpamirror version).
5. Restart emacs, Ah~~！It works!

I change back to spacemacs, errors reproduced. So should there are some conflicts between this package and spacemacs? I would try to find it out then. Thanks!
